### ISSUE TITLE
Problems with the ontologies, and some glyph names

### ISSUE THREAD

**Issue by adrienrougny** [sbgn/process-descriptions#207]
The ontologies in section 3.3 "The conceptual model" mix glyphs and concepts, which doesn't seem correct.
Indeed, in those ontologies, nodes are classes of glyphs, and arcs _is_a_ relationships shared among concepts (and not glyphs): For example, in Figure 3.1, the **Process** and **Omitted process** classes refer to the _process_ and the _omitted process_ glyphs respectively (as they are subclasses of the **SBGNGlyph** class), and the **Omitted process** class is a subclass of the **Process class**. In a given map, all instances of the _omitted process_ glyph are not instances of the _process_ glyph (simply because the glyphs are different), but in terms of biological concepts, all instances of omitted processes are instances of processes (i.e. a process is an omitted process). Hence the relationship **Omitted process** _is_a_ **Process** only holds if those classes refer to the biological concepts (and not to the glyphs used to represent these concepts). 
In general, I believe an ontology organizing the relationships shared between glyphs is not interesting, because its structure would have to be horizontal: all classes of processes (referring to the _process_ glyph, the _omitted process_ glyph, the _association_ glyph, etc.) would have to be sister classes (i.e. share the same parent class); and it would be the same for all other types of glyphs (modulations, EPNs, etc.). That is of course not the case for ontologies organizing the represented concepts (see SBO).

Another related issue is whether the _process_ glyph can replace any other type of process glyph (the _omitted process_ glyph, the _dissociation_ glyph, etc.), i.e. whether the _process_ glyph can be used to represent any process (and possibly losing some information doing so), or not. The same question arises for the _modulation_ glyph. I also believe that, interestingly, it is relevant for the _unspecified entity_ glyph. The alternatives are illustrated in the following [PDF](https://github.com/sbgn/process-descriptions/files/2061026/issue.pdf).

Finally, it would be good to rename the _process_ glyph and the _modulation_ glyph: currently, when talking of "process grlyphs", we do not know if we're talking about "all instances of the _process_ glyph" or all instances of glyphs that represent processes (i.e. all instances of the _process_ glyph, together with those of the _dissociation_, _association_, ..., glyphs). In Stuart's version of the specification L1V2, the _process_ glyph was renamed to the _generic process_ glyph. The same could be done for the _modulation_ glyph.

**Comment by amazein** [comment]
This should be discussed via [sbgn-discuss], voted on etc. For L1V2 we should act on what we already have so we can do it quickly. If you would like to change names of the glyphs in L1V2 we should start the discussion via [sbgn-discuss] soon. I think we can postpone this. I agree, there are inconsistencies there, they should be fixed, but they are not critical for the use of the notation.

On the topic itself, I understand the motivation for renaming the _process_ glyph. On the other hand, _generic process_ assumes specific processes. if in the future we remove _association_ and _dissociation_ glyphs, there would not be any specific processes over which we would have _generic process_. This might not be a problem though.

For _modulation_, _stimulation_, _catalysis_ and _inhibition arcs_, the term could be _regulatory arcs_?

**Comment by Adrienrougny** [comment]
I agree it's no urgent matter and it can be postponed (that's why I hadn't assign any milestone to it), and even discussed later. I just had to write it before forgetting it.

For the _generic_ process, even without the _association_ and _dissociation_,  wouldn't the _omitted process_ and _uncertain process_ still represent specific kind of processes?

**Comment by amazein** [comment]
Yes, _generic process_ sounds good, even without _association_ and _dissociation_.

**Comment by hasanbalci** [comment]
Apparently, there are a few questions to answer in this issue:
- Differentiation of glyphs and concepts in Section 3.3.
- Can the process glyph replace any other type of process glyphs?
- Renaming the process glyph and modulation glyph. Currently the term "generic process" is used in section 2.6 Process Nodes.

**Comment by draeger** [comment]
While I see the intentions behind it, I am not yet fully convinced. For instance, the process node can be used instead of any of those specific ones. The same is true for the modulation arc, and in the end also for the label. Please also keep in mind the consequences with respect to related standards, such as the SBML layout extension if any. Most symbols also correspond to specific SBO terms, which are are also hierarchically organized. When one is changed, the other should match it as well.

**Comment by adrienrougny** [comment]
I will split the conceptual model into two figures to solve the issue of mixing glyphs and concepts in the ontology. If they are ok we integrate them in the spec. The other issues related to the conceptual model will be solved later.

**Comment by adrienrougny** [comment]
[conceptual_model.gv.pdf](https://github.com/user-attachments/files/17111270/conceptual_model.gv.pdf)
[glyph_model.gv.pdf](https://github.com/user-attachments/files/17111271/glyph_model.gv.pdf)
Two draft models for the concepts and the glyphs, represented using Visual OWL (notation for representing owl ontologies).
The models could be split or represented with a different layout (they are made with graphviz, so other types of layouts can easily be applied to them). TODO: represent abstract class with italics.

**Comment by adrienrougny** [comment]
TODO Adrien: put the source files

**Comment by adrienrougny** [comment]
[conceptual_model.gv.txt](https://github.com/user-attachments/files/17799889/conceptual_model.gv.txt)
[glyph_model.gv.txt](https://github.com/user-attachments/files/17799890/glyph_model.gv.txt)

**Comment by hasanbalci** [comment]
A few points to consider on glyph model figure:
- Maybe we can group multimer nodes under a parent node named "multimer node" (similar to subunits, and in this way, we can also reduce the number of nodes at level 3 for a better layout.)
- Should we use "simple chemical multimer" or "multimer of simple chemicals"? The first one seems more appropriate to me.
- Should we add "clone marker" as a child node of "auxiliary unit/decorator"?
- "equivalence operator" can be included under the "logical operator node" as we categorize it as logical operator in the spec.
- "terminal" -> "submap terminal"

**Comment by cannin** [comment]
Imperfect, but less wide. +1 on grouping multimers. 

![glyph_model_with_layout_yed](https://github.com/user-attachments/assets/db40852c-c523-4aa3-9085-66455fe700ae)
[glyph_model_with_layout_yed.graphml.txt](https://github.com/user-attachments/files/17840023/glyph_model_with_layout_yed.graphml.txt)

**Comment by adrienrougny** [comment]
Nice graph!

- Also ok for grouping multimers.
- I prefer multimer of simple chemicals (it's the Systems BIology Ontology term), but in the spec the glyph is name "simple chemical multimer", so maybe we should stick with it (or change the name in the spec)
- Yes clone marker could be a child of auxiliary unit, although it is not really a node
- Ok for having the equivalence operator as a logical operator node, since it is like that in the spec. But the equivalence operator is actually not a logical operator in the logical sense, so we could think of modifying the spec and having the equivalence operator as an individual node
- Terminal->Submap terminal ok

On the graph, we could highlight all abstract classes, either with a specific color (like SBGN node, glyph, arc), or by italicizing the label, for example.

**Comment by hasanbalci** [comment]
I tried to improve layout and applied some of the points above.
![glyph_model_with_layout_yed](https://github.com/user-attachments/assets/a8fff270-1fdc-4ac1-949a-5f14455d8b82)


[glyph_model_with_layout_yed_v2.graphml.txt](https://github.com/user-attachments/files/17848203/glyph_model_with_layout_yed_v2.graphml.txt)

**Comment by hasanbalci** [comment]
I tried to improve layout and applied some of the points above.
![glyph_model_with_layout_yed_v2](https://github.com/user-attachments/assets/0c94b893-ce55-406d-9dfe-e8e86f49e3dd)
[glyph_model_with_layout_yed_v2.graphml.txt](https://github.com/user-attachments/files/17848374/glyph_model_with_layout_yed_v2.graphml.txt)

**Comment by adrienrougny** [comment]
Looks great!

**Comment by adrienrougny** [comment]
Could we add a subunit multimer node for symmetry?

**Comment by adrienrougny** [comment]
Looking at the model, we need to harmonize the names for multimers (as pointed out above):
Currently, we have:
* names of multimer glyphs as "macromolecule multimer", "simple chemical multimer", ...
* names of multimer subunits as "multimer of macromolecules subunit", "multimer of simple chemicals subunit", ...
* SBO terms mentioned in the spec as "multimer of macromolecules", "multimer of simple chemicals", ...
I would harmonize based on the SBO terms, that is:
* change the names of the multimer glyphs to "multimer of macromolecules", "multimer of simple chemicals"
and:
* change the name of the abstract "mutimer of subunits" class to "multimer subunit"

**Comment by adrienrougny** [comment]
[sbgn_glyph_model.graphml.txt](https://github.com/user-attachments/files/17933412/sbgn_glyph_model.graphml.txt)
[sbgn_glyph_model.pdf](https://github.com/user-attachments/files/17933414/sbgn_glyph_model.pdf)

**Comment by hasanbalci** [comment]
@adrienrougny I agree, otherwise it seems inconsistent. I uploaded this last one to the overleaf. We may consider changing the naming of these multimers in the spec as well, as you suggested before. 

Ugur and I also considered changing the following section headers in the spec to be consistent with the other headers: Logical operators -> Logical operator nodes and Encapsulation -> Encapsulation node. Let's discuss this in the meeting.
